kaggle_environments/envs/halite/helpers.py

```diff
@@ -0,0 +1,781 @@
+        return self._next_action
+
+    @next_action.setter
+    def next_action(self, value: Optional[ShipyardAction]) -> None:
+        """Sets the action that will be executed by this shipyard when Board.next() is called (when the current turn ends)."""
+        self._next_action = value
+
+    @property
+    def observation(self) -> int:
+        """Converts a shipyard back to the normalized observation subset that constructed it."""
+        return position_to_index(self.position, self._board.configuration.size)
+
+
+class Player:
+    def __init__(self, player_id: PlayerId, halite: int, shipyard_ids: List[ShipyardId], ship_ids: List[ShipId], board: 'Board') -> None:
+        self._id = player_id
+        self._halite = halite
+        self._shipyard_ids = shipyard_ids
+        self._ship_ids = ship_ids
+        self._board = board
+
+    @property
+    def id(self) -> PlayerId:
+        return self._id
+
+    @property
+    def halite(self) -> int:
+        return self._halite
+
+    @property
+    def shipyard_ids(self) -> List[ShipyardId]:
+        return self._shipyard_ids
+
+    @property
+    def ship_ids(self) -> List[ShipId]:
+        return self._ship_ids
+
+    @property
+    def shipyards(self) -> List[Shipyard]:
+        """Returns all shipyards owned by this player."""
+        return [
+            self._board.shipyards[shipyard_id]
+            for shipyard_id in self.shipyard_ids
+        ]
+
+    @property
+    def ships(self) -> List[Ship]:
+        """Returns all ships owned by this player."""
+        return [
+            self._board.ships[ship_id]
+            for ship_id in self.ship_ids
+        ]
+
+    @property
+    def is_current_player(self) -> bool:
+        """Returns whether this player is the current player (generally if this returns True, this player is you)."""
+        return self.id is self._board.current_player_id
+
+    @property
+    def next_actions(self) -> Dict[str, str]:
+        """Returns all queued ship and shipyard actions for this player formatted for the halite interpreter to receive as an agent response."""
+        ship_actions = {
+            ship.id: ship.next_action.name
+            for ship in self.ships
+            if ship.next_action is not None
+        }
+        shipyard_actions = {
+            shipyard.id: shipyard.next_action.name
+            for shipyard in self.shipyards
+            if shipyard.next_action is not None
+        }
+        return {**ship_actions, **shipyard_actions}
+
+    @property
+    def observation(self):
+        """Converts a player back to the normalized observation subset that constructed it."""
+        shipyards = {shipyard.id: shipyard.observation for shipyard in self.shipyards}
+        ships = {ship.id: ship.observation for ship in self.ships}
+        return [self.halite, shipyards, ships]
+# endregion
+
+
+class Board:
+    def __init__(
```

Comment:
The constructor is basically just used to deserialize the raw observation and configuration received from the halite interpreter into the richer Board model that we'll make use of in the next() method down below.